### DISCUSSION TITLE
A problem of white halo and not-smooth in edge of displaced image

### DISCUSSION THREAD

**Discussion by prettyblueberry** [ImageMagick/ImageMagick6#321] [General]
### ImageMagick version

6.9.11

### Operating system, version and so on

Ubuntu 20.04

### Description

Dear everyone,

Now I'm trying to put an image on a pillow.
But the result has white halo and not smooth edge. The result image shows white halo with black background.
I have read some blogs so I already added some different code pieces from blogs( including [http://www.imagemagick.org/discourse-server/viewtopic.php?t=22890](http://www.imagemagick.org/discourse-server/viewtopic.php?t=22890), in the blog, @snibgo and @fmw42 discussed about similar problem so I tried but not success.)

```
#!/bin/bash

convert 0_temp.png[1000x1000+0+0] -colorspace gray -auto-level 1_displacement_map.png
convert 0_design.png -resize 1000x1000 -filter Lanczos mid_resized.png
convert mid_resized.png 1_displacement_map.png -alpha set -virtual-pixel transparent -compose displace -set option:compose:args -5x-5 -composite 1_displaced.png
convert 1_displaced.png -channel matte -separate 1_logo_displace_mask.png
convert  "mid_resized.png"  -bordercolor transparent -border 12x12 temp.png
convert 0_temp.png -colorspace gray -blur 5x500 -auto-level displace_map.png
convert 0_temp.png  displace_map.png -alpha set -virtual-pixel transparent -compose displace -set option:compose:args -5x-5 -composite displaced_logo.png
convert "displace_map.png" -colorspace gray -auto-level light_map.png
convert light_map.png -alpha copy -channel A -negate +channel -channel rgb -evaluate set 0 -channel rgba -alpha on PNG32:light_map.png
convert "displaced_logo.png" -channel matte -separate logo_displace_mask.png
convert 1_displaced.png light_map.png -compose Multiply 1_logo_displace_mask.png -composite light_map_logo.png
convert light_map_logo.png light_map.png -compose Multiply 1_logo_displace_mask.png -composite light_map_logo.png
convert 0_temp.png light_map_logo.png -gravity southeast -geometry +0+0 -composite result.png
```

"0_temp.png" and "0_design.png" are input files and "result.png" is output.

![0_pillow](https://github.com/ImageMagick/ImageMagick/assets/121964338/5782201c-0acb-469e-9d47-29ea8a0875f3)
![0_design](https://github.com/ImageMagick/ImageMagick/assets/121964338/3523a698-4d23-470b-943a-2a1c82984841)
![result](https://github.com/ImageMagick/ImageMagick/assets/121964338/c9477167-c175-4c44-9033-d2069af1250a)

**Comment by fmw42** [comment]
Sorry, but I do not see any halo. The image may not be as sharp due to resizing and displacing. But I don't see anything unexpected.

For this image, you do not need all that processing. Just composite it on the pillow.
